molecular/src/main/kotlin/com/hartwig/actin/molecular/orange/CharacteristicsExtraction.kt

```diff
@@ -26,17 +27,17 @@ object CharacteristicsExtraction {
         return MolecularCharacteristics(
             purity = purple.fit().purity(),
             ploidy = purple.fit().ploidy(),
-            predictedTumorOrigin = determinePredictedTumorOrigin(record.cuppa()),
+            predictedTumorOrigin = determinePredictedTumorOrigin(record.cuppa(), record),
             microsatelliteStability = determineMicrosatelliteStability(purple.characteristics()),
             homologousRecombination = determineHomologousRecombination(record.chord()),
             tumorMutationalBurden = determineTumorMutationalBurden(purple.characteristics()),
             tumorMutationalLoad = determineTumorMutationalLoad(purple.characteristics())
         )
     }
 
-    private fun determinePredictedTumorOrigin(cuppa: CuppaData?): PredictedTumorOrigin? {
+    private fun determinePredictedTumorOrigin(cuppa: CuppaData?, record: OrangeRecord): PredictedTumorOrigin? {
```

Comment:
you don't need to pass the whole orange record through here. The cuppaData already has the mode in it